meson.build

```diff
@@ -0,0 +1,60 @@
+project('tmp', 'cpp', version: '2.0', meson_version: '>=1.3')
+
+include_directory = include_directories('include')
+cxx = meson.get_compiler('cpp')
+
+std_fs_dep = cxx.find_library('stdc++fs', required: false)
+
+tmp = library(
+  'tmp',
+  'src/create.cpp',
+  'src/entry.cpp',
+  'src/file.cpp',
+  'src/directory.cpp',
+  'src/move.cpp',
+  install: true,
+  include_directories: include_directory,
+  gnu_symbol_visibility: 'hidden',
+  version: meson.project_version(),
+  dependencies: [std_fs_dep],
+  cpp_shared_args: ['-DTMP_BUILDING_DLL'],
```

Comment:
... which means you can choose to pass this define only if `host_machine.system() == 'windows'`, and therefore allow linux / macOS to share objects for both static/shared.